ngraph/core/CMakeLists.txt

```diff
@@ -33,8 +33,13 @@ source_group("include" FILES ${PUBLIC_HEADERS})
 
 configure_file(include/ngraph/version.in.hpp include/ngraph/version.hpp)
 
+if (NGRAPH_STATIC_LIBRARY)
+# Create static library
+    add_library(ngraph STATIC ${LIBRARY_SRC} ${PUBLIC_HEADERS})
```

Comment:
From previous PR:

@ilya-lavrenov 
>> I think we can just add:
>> `add_library(ngraph ${LIBRARY_SRC} ${PUBLIC_HEADERS})`
>> By default it will be SHARED, but we pass cmake -DBUILD_SHARED_LIBS=OFF (see  https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html), cmake sets target type as STATIC

@emmanuelattia-philips 
> Linked to ngraph/CMakeLists.txt conversation
> It seems that by default BUILD_SHARED_LIBS is OFF, i can change that
>
> But some other of the libraries already rely on that flag (onnx, onnx_importer, onnx_proto). Should i put STATIC explicit to them ? otherwise they will end up being dynamic as well (unlike today)
>
> I tried to compile the master version passing -DBUILD_SHARED_LIBS=ON and it has side effects, for instance it compiles onnx as a dll and fails later because it's in no export target.
>
> So i'm afraid the CMake process relies on the fact that BUILD_SHARED_LIBS=OFF even when building shared versions of openvino's libraries.

Unfortunately ONNX is a third-party dependency. Can we force disable BUILD_SHARED_LIBS for some components (if we want to use this flag for static/dynamic build)?

Reply:
I think i was able to solve that (see https://github.com/openvinotoolkit/openvino/pull/2892/commits/044015b7b1bf402c081e317f1852d4dffcf63da1)
Now my problem is that by default BUILD_SHARED_LIBS=OFF and if i want to use this option instead of NGRAPH_STATIC_LIBRARY, I have to define the option explicitely somewhere so it's ON by default.

Where should i do that ? in /cmake/features.cmake ?

Reply:
> Where should i do that ? in /cmake/features.cmake ?

Right.

Reply:
What would be the description since for this first PR it would not be accurate that it would build all the openvino's library as static but just ngraph. I propose "Build as much static libraries as possible" ?

Reply:
I think the description is OK :)

Reply:
Ok now it's using BUILD_SHARED_LIBS